### PR TITLE
refactor: ledger & anoncreds typing, part 2

### PR DESCRIPTION
**Pull request by mirgee**
Continuation of #1116.

### CODE REVIEW

File: aries/aries_vcx_core/src/anoncreds/anoncreds/mod.rs [hunk 1]
```diff
@@ -1109,28 +1126,23 @@ impl BaseAnonCreds for Anoncreds {
...
-        cred_rev_id: &str,
+        cred_rev_id: u32,
     ) -> VcxCoreResult<String> {
-        let mut rev_reg_def_json: Value = serde_json::from_str(rev_reg_def_json)?;
-        let rev_reg_def_id = rev_reg_def_json["id"].as_str().unwrap().to_string();
-        let cred_def_id = rev_reg_def_json["credDefId"].as_str().unwrap();
-
+        let cred_def_id = rev_reg_def_json.cred_def_id.to_string();
+        let max_cred_num = rev_reg_def_json.value.max_cred_num;
+        let rev_reg_def_id = rev_reg_def_json.id.to_string();
```
Comment (Patrik-Stas) [inline aries/aries_vcx_core/src/anoncreds/anoncreds/mod.rs:1137]: ❤️

File: aries/aries_vcx_core/src/anoncreds/anoncreds/mod.rs [hunk 2]
```diff
@@ -1079,17 +1096,17 @@ impl BaseAnonCreds for Anoncreds {
...
-            cred_def_id.split(':').next().unwrap().into(),
-        );
-        let cred_def: CredentialDefinition = serde_json::from_str(&cred_def_json.to_string())?;
-        let credential_offer: CredentialOffer = serde_json::from_str(cred_offer_json)?;
+        // let mut cred_def_json: Value = serde_json::from_str(cred_def_json)?;
+        // let cred_def_id = cred_def_json["id"].as_str().unwrap().to_string();
+        // cred_def_json.as_object_mut().unwrap().insert(
+        //     "issuerId".to_owned(),
+        //     cred_def_id.split(':').next().unwrap().into(),
+        // );
```
Comment (Patrik-Stas) [inline aries/aries_vcx_core/src/anoncreds/anoncreds/mod.rs:1107]: can be deleted?

File: aries/aries_vcx_core/src/anoncreds/anoncreds/mod.rs [hunk 3]
```diff
@@ -1460,39 +1448,25 @@ impl BaseAnonCreds for Anoncreds {
...
-
-        if let Err(err) = &res_rev_reg_delta {
-            warn!(
-                "get_rev_reg_delta >> Unable to deserialize rev_reg_delta cache for rev_reg_id: \
-                 {}, error: {}",
-                rev_reg_id, err
-            );
-        }
-
-        Ok(res_rev_reg_delta.ok().flatten())
```
Comment (Patrik-Stas) [inline aries/aries_vcx_core/src/anoncreds/anoncreds/mod.rs:1477]: love to see it gone

**Review by Patrik-Stas — APPROVED** [review]
minor comments, but LGTM 🟢

File: aries/aries_vcx/src/common/primitives/revocation_registry_delta.rs [hunk 4]
```diff
@@ -3,8 +3,8 @@ use crate::errors::error::prelude::*;
 #[derive(Clone, Deserialize, Debug, Serialize, Default)]
 pub struct RevocationRegistryDelta {
     value: RevocationRegistryDeltaValue,
-    #[serde(rename = "ver")]
-    version: String,
+    // #[serde(rename = "ver")]
+    // version: String,
```
[the remark this answers is not in the document]
Comment (mirgee) [reply aries/aries_vcx/src/common/primitives/revocation_registry_delta.rs:7]: Thank you for catching this, removed in the next PR

File: aries/aries_vcx/src/common/primitives/revocation_registry.rs [hunk 5]
```diff
@@ -220,14 +235,19 @@ impl RevocationRegistry {
         wallet: &impl BaseWallet,
         anoncreds: &impl BaseAnonCreds,
         ledger: &impl AnoncredsLedgerRead,
-        cred_rev_id: &str,
+        cred_rev_id: u32,
     ) -> VcxResult<()> {
         let rev_reg_delta_json = ledger
-            .get_rev_reg_delta_json(&self.rev_reg_id, None, None)
+            .get_rev_reg_delta_json(&self.rev_reg_id.to_string().try_into()?, None, None)
```
[the remark this answers is not in the document]
Comment (mirgee) [reply aries/aries_vcx/src/common/primitives/revocation_registry.rs:241]: > future task: perhaps get rid of impl_anoncreds_object_identifier macro? (used for generation of RevocationRegistryDefinitionId)
Yes, I think it is already a TODO left over from the last review :)

> Stating obvious, but pub rev_reg_id: String on RevocationRegistry should be typed, so we don't have to do these conversion in functions it implements. Probably something for the part3+ iteration.
All of these "primitives" deserve rework after introducing the `anoncreds_types` crate. Most of the fields of this struct are duplicated in `RevocationRegistryDefinition` in `anoncreds_types`.

File: aries/aries_vcx_core/src/anoncreds/anoncreds/mod.rs (hunk 2 above)
Comment (mirgee) [reply aries/aries_vcx_core/src/anoncreds/anoncreds/mod.rs:1107]: Thank you for catching this, removed in the next PR

File: aries/aries_vcx_core/src/anoncreds/anoncreds/mod.rs (hunk 3 above)
Comment (mirgee) [reply aries/aries_vcx_core/src/anoncreds/anoncreds/mod.rs:1477]: Me too :) This is just the beginning (hopefully)

**Review by Patrik-Stas — APPROVED** [review]

**Pull request merged**